recastjs/src/recastjs.cpp

```diff
@@ -0,0 +1,577 @@
+    {
+        Log("buildNavigation: Could not create solid heightfield.");
+        return ;
+    }
+
+    float *verts = new float[mTriangles.size()*3];
+    int nverts = mTriangles.size();
+    for (unsigned int i =0;i<mTriangles.size();i++)
+    {
+        verts[i*3+0] = mTriangles[i].x;
+        verts[i*3+1] = mTriangles[i].y;
+        verts[i*3+2] = mTriangles[i].z;
+    }
+    int ntris = mTriangles.size()/3;
+    int *tris = new int[mTriangles.size()];
+    for (unsigned int i = 0;i<mTriangles.size();i++)
+        tris[i] = mTriangles.size()-i-1;
+
+
+    //InputGeom* m_geom;
+    // Allocate array that can hold triangle area types.
+    // If you have multiple meshes you need to process, allocate
+    // and array which can hold the max number of triangles you need to process.
+    unsigned char*m_triareas = (unsigned char*)malloc(ntris);
+    
+    
+    // Find triangles which are walkable based on their slope and rasterize them.
+    // If your input data is multiple meshes, you can transform them here, calculate
+    // the are type for each of the meshes and rasterize them.
+    memset(m_triareas, RC_WALKABLE_AREA, ntris*sizeof(unsigned char));
+
+
+    //rcMarkWalkableTriangles(m_ctx, m_cfg.walkableSlopeAngle, verts, nverts, tris, ntris, m_triareas);
+    rcRasterizeTriangles(m_ctx, verts, nverts, tris, m_triareas, ntris, *m_solid, m_cfg.walkableClimb);
+
+    free(m_triareas);
+    
+    //
+    // Step 3. Filter walkables surfaces.
+    //
+    
+    // Once all geoemtry is rasterized, we do initial pass of filtering to
+    // remove unwanted overhangs caused by the conservative rasterization
+    // as well as filter spans where the character cannot possibly stand.
+    
+    rcFilterLowHangingWalkableObstacles(m_ctx, m_cfg.walkableClimb, *m_solid);
+    rcFilterLedgeSpans(m_ctx, m_cfg.walkableHeight, m_cfg.walkableClimb, *m_solid);
+    rcFilterWalkableLowHeightSpans(m_ctx, m_cfg.walkableHeight, *m_solid);
+    
+
+    //
+    // Step 4. Partition walkable surface to simple regions.
+    // 
+
+    // Compact the heightfield so that it is faster to handle from now on.
+    // This will result more cache coherent data as well as the neighbours
+    // between walkable cells will be calculated.
+    rcCompactHeightfield* m_chf;
+    m_chf = rcAllocCompactHeightfield();
+    if (!m_chf)
+    {
+        Log("buildNavigation: Out of memory 'chf'.");
+        return ;
+    }
+    
+    if (!rcBuildCompactHeightfield(m_ctx, m_cfg.walkableHeight, m_cfg.walkableClimb, *m_solid, *m_chf))
+    {
+        Log("buildNavigation: Could not build compact data.");
+        return ;
+    }
+    
+    if (!m_keepInterResults)
+    {
+        rcFreeHeightField(m_solid);
+        m_solid = 0;
+    }
+        
+    // Erode the walkable area by agent radius.
+    if (!rcErodeWalkableArea(m_ctx, m_cfg.walkableRadius, *m_chf))
+    {
+        Log("buildNavigation: Could not erode.");
+        return ;
+    }
+
+    // (Optional) Mark areas.
```

Comment:
I would recommend either removing the commented code or adding a comment explaining why the code is commented out.

Reply:
Removed